micro-rdk/src/esp32/pin.rs

```diff
@@ -33,6 +34,17 @@ fn install_gpio_isr_service() -> Result<(), BoardError> {
     Ok(())
 }
 
+// Since C macros are not usable in Rust via FFI, we cannot GPIO_IS_VALID_DIGITAL_IO_PAD from
+// ESP-IDF, so we must replicate the logic from that function here. If we do not validate,
+// PinDriver::input_output will panic because esp-idf-hal does not perform the same check
+unsafe fn validate_pin_as_io(pin: i32) -> Result<(), BoardError> {
```

Comment:
I think this convention works better for functions that return booleans

Reply:
Sure, but on the other hand, since it returns `Result<{}`, in practice usage is going to look like:
`is_valid_whatever()?` if you don't care to inspect the error type and just want to propagate with `?` I don't feel that strongly about it, so up to you. I do think it should say `gpio` rather than `io` though.

Reply:
fair, I'll change